Sources/Dependencies/DependencyValues.swift

```diff
@@ -349,60 +351,62 @@ private final class CachedValues: @unchecked Sendable {
 
 // NB: We cannot statically link/load XCTest on Apple platforms, so we dynamically load things
 //     instead and we limit this to debug builds to avoid App Store binary rejection.
-#if !canImport(ObjectiveC)
-  import XCTest
-#endif
+#if DEBUG
```

Comment:
I'm not sure what the reason was to remove the `DEBUG` guard but if it needs to be removed, perhaps a better solution would be something like

```
#if DEBUG || !os(Linux)
```

or maybe simply

```
#if !os(Linux)
```

?

I'm not sure what the block does exactly so it's hard to judge under what conditions it needs to be compiled in and what its absence means, functionality-wise.

Reply:
Hi @finestructure, we are wanting to remove the `#if DEBUG` because it is a bummer to have different behavior for debug vs release builds in tests. It basically means you cannot run your test code in release mode if it uses this library. The same is true of our xctest-dynamic-overlay library and we are hoping to remove the `#if DEBUG`s from that too.

It seems like we can instead wrap the entire testing logic in a `#if canImport(XCTest)` since that will be `false` when built with `--static-stdlib`, and it’s already not possible to run swift test `--static-stdlib` so we aren’t losing anything there.

What do you think of making that change instead? Or if you don't have time we can take care of it.

Reply:
I see, that makes sense! Let me update that real quick but if it's still not quite right please feel free to mess with the PR!

Reply:
Great, I just pushed some fixes and I added `swift build -static-stdlib` to CI so that we can catch these things in the future.